doc/contributing/docs/git.rst

```diff
@@ -0,0 +1,80 @@
+Git workflow
+============
+
+Branching
+---------
+
+Use one branch for a single task, unless you're fixing typos or markup on several pages.
+Long commit histories are hard to manage and sometimes end up stale.
+
+Start a new branch from the last commit on ``latest``.
+Make sure to update your local version of ``latest`` with ``git pull``.
+Otherwise, you may have to rebase later.
+
+Name your branch so it's clear what you're doing. Examples:
+
+*   ``short-issue-description``
+*   ``gh-1234-short-issue-description``
+*   ``your-github-handle/short-issue-description``
+
+Linking issues and PRs
+----------------------
+
+When a PR is linked to an issue:
+
+*   You can go from the issue straight to the PR by clicking the link in the right column.
+*   The issue will be automatically closed when you close the PR.
+
+Specify the issue(s) you want to close in the description of your PR. GitHub will connect them if you use specific
+`keywords <https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword>`__.
+Here are some of them:
+
+*   Closes #1234
+*   Resolves #1234
+*   Fixes #1234
+
+If your PR closes more than one issue, add such a line for each issue.
+
+Commit messages
+---------------
+
+*   One-line commit messages are sufficient for documentation changes. Try keeping it 50 characters or less.
```

Comment:
There can be a second line with "Resolves #...". In fact, a commit is often made before opening a PR.

Reply:
When you squash commits at merge, the resulting message will be a sum of commit messages. So the first commit should have the "resolves" string. Otherwise, there's a risk that this line won't be in the resulting commit.

Reply:
So, the best practice is to write a good commit message first, and then to open a PR

Reply:
> In fact, a commit is often made before opening a PR.
Do you mean this process: First you implement changes locally, then you push the branch with the commit, which prompts to create the PR?

>  the first commit should have the "resolves" string
You mean that the first commit message in the branch must have a second line with the "Resolves" string? Currently we don't do that, but we include the "resolves" string in the PR description when creating the PR. Do you think we really need to introduce this practice?